### PR TITLE
add Particular.CodeRules and remove R# settings

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

**Comment by HEskandari** [comment]
@adamralph due to the [other PR](https://github.com/Particular/NServiceBus.PowerShell/pull/63) (changing to the new csproj format), this will need to be reworked. I'll have a look after that is merged.

**Comment by adamralph** [comment]
@HEskandari I'll rebase this. The conversion to new csproj will make this _much_ simpler.

**Comment by adamralph** [comment]
Although the changes to add Particular.CodeRules are now much simpler, the analysers actually _run_ now, so this now includes all the required style fixes too.

File: src/NServiceBus.PowerShell.Tests/NServiceBus.PowerShell.Tests.csproj [hunk 1]
```diff
@@ -11,6 +11,7 @@
 
   <ItemGroup>
     <PackageReference Include="NUnit" Version="3.5.0" />
```
Comment (danielmarbach) [inline src/NServiceBus.PowerShell.Tests/NServiceBus.PowerShell.Tests.csproj:13]: More an observation and probably unrelated to this PR but why doesn't this reference the MS Test SDK and the unit adapters?

File: src/NServiceBus.PowerShell/Cmdlets/RemoveMSMQWorker.cs [hunk 2]
```diff
@@ -22,9 +22,8 @@ public class RemoveMSMQWorker : PSCmdlet
 
         [Parameter(Mandatory = true, Position = 2, HelpMessage = "The Distributor address.", ValueFromPipeline = true)]
         public bool TransactionalDistributorQueue { get; set; }
-       
-        // ReSharper enable  MemberCanBePrivate.Global
 
+        // ReSharper enable  MemberCanBePrivate.Global
```
Comment (danielmarbach) [inline src/NServiceBus.PowerShell/Cmdlets/RemoveMSMQWorker.cs:26]: Ok so from now on you are leaving these in?

File: src/NServiceBus.PowerShell/Dtc/DtcSetup.cs [hunk 3]
```diff
@@ -119,7 +119,7 @@ bool DoesSecurityConfigurationRequireRestart(bool doChanges, string PortRange =
                     PortRange
                 };
 
-                if (Array.IndexOf((string[]) hklm.ReadValue(rpcKeyName, RpcPortsKey, new string[]{}, true), PortRange) >= 0)
+                if (Array.IndexOf((string[])hklm.ReadValue(rpcKeyName, RpcPortsKey, new string[] { }, true), PortRange) >= 0)
```
Comment (danielmarbach) [inline src/NServiceBus.PowerShell/Dtc/DtcSetup.cs:122]: Could we use `Array.Empty` here or is that not supported by the TFM?

File: src/NServiceBus.PowerShell/Msmq/MsmqSetup.cs [hunk 4]
```diff
@@ -10,7 +10,7 @@ namespace NServiceBus.PowerShell
     using System.ServiceProcess;
     using System.Text;
     using Helpers;
-    
+
```
Comment (danielmarbach) [inline src/NServiceBus.PowerShell/Msmq/MsmqSetup.cs:13]: This whole file is hard to review. So I did sanity check some of the conditions and check whether any of the changes did swap boolean logic and it seems fine.

**Review by danielmarbach — APPROVED** [review]

File: src/NServiceBus.PowerShell.Tests/NServiceBus.PowerShell.Tests.csproj (hunk 1 above)
Comment (adamralph) [reply src/NServiceBus.PowerShell.Tests/NServiceBus.PowerShell.Tests.csproj:13]: 🤷

File: src/NServiceBus.PowerShell/Cmdlets/RemoveMSMQWorker.cs (hunk 2 above)
Comment (adamralph) [reply src/NServiceBus.PowerShell/Cmdlets/RemoveMSMQWorker.cs:26]: Yep, as discussed.

File: src/NServiceBus.PowerShell/Dtc/DtcSetup.cs (hunk 3 above)
Comment (adamralph) [reply src/NServiceBus.PowerShell/Dtc/DtcSetup.cs:122]: Unfortunately [only `net46` onwards](https://docs.microsoft.com/en-us/dotnet/api/system.array.empty), and this project is `net452`.

File: src/NServiceBus.PowerShell/Msmq/MsmqSetup.cs (hunk 4 above)
Comment (adamralph) [reply src/NServiceBus.PowerShell/Msmq/MsmqSetup.cs:13]: Yeah, the white space was totally out on this one. IIRC I only applied the automatic fix provided by the analyser but I guess that reformatted the whole file.

**Pull request merged**